server/polar/issue/endpoints.py

```diff
@@ -182,37 +182,23 @@ async def lookup(
 
         client = get_polar_client()
 
-        try:
-            res = (
-                await github_organization_service.sync_external_org_with_repo_and_issue(
-                    session,
-                    client=client,
-                    org_name=url.owner,
-                    repo_name=url.repo,
-                    issue_number=url.number,
-                )
-            )
-            org, repo, tmp_issue = res
-        except ResourceNotFound:
-            raise HTTPException(
-                status_code=404,
-                detail="Issue by external_url not found",
-            )
+        res = await github_organization_service.sync_external_org_with_repo_and_issue(
+            session,
+            client=client,
+            org_name=url.owner,
+            repo_name=url.repo,
+            issue_number=url.number,
+        )
+        org, repo, tmp_issue = res
 
         # get for return
         issue = await issue_service.get_loaded(session, tmp_issue.id)
         if not issue:
-            raise HTTPException(
-                status_code=404,
-                detail="Issue not found",
-            )
+            raise ResourceNotFound("Issue not found")
```

Comment:
Similarly, for logic directly in controllers where we manually raise `HTTPException`, it could be interesting to move to this common pattern.